src/main/scala/au/com/agiledigital/toolform/reader/ProjectReader.scala

```diff
@@ -34,13 +42,44 @@ object ProjectReader {
           val config = simpleConfig.resolve()
 
           val projectResult = loadConfig[Project](config)
-          resultOrCollectReadErrors(projectResult)
+
+          for {
+            collectedReadErrors    <- resultOrCollectReadErrors(projectResult)
+            validatedProjectResult <- validateProjectResult(collectedReadErrors)
+          } yield validatedProjectResult
+
         case Failure(e) =>
-          Left(ToolFormError(s"Failed to parse project configuration [$projectDefinitionFile]: [${e.getMessage}]"))
+          Left(NonEmptyList.of(ToolFormError(s"Failed to parse project configuration [$projectDefinitionFile]: [${e.getMessage}]")))
+      }
+    }
+
+  private def validateProjectResult(project: Project): Either[NonEmptyList[ToolFormError], Project] =
```

Comment:
Prefer `validatedProject` since this is not mutating. Reserve verb names for mutators / side-effecting functions.

Reply:
Fixed